### ISSUE TITLE
Weird behavior redefining constructors

### ISSUE THREAD

**Issue by lauraharker** [google/closure-compiler#2928]
This doesn't cause any type warnings. The type system merges the properties of the first and second declaration of `C` into a single type.
``` javascript
var C = class {
  f() {}
}; 
C = class {
  g() {}
};
var n = new C();

n.f();
n.g();
```

There are certain situations where we do get errors for redefining classes. `FunctionTypeBuilder` throws an error when the constructor signatures are different. 

If the second class has a supertype, the supertype doesn't get put into the actual FunctionType, but it does result in a $jscomp.inherits() call that causes an error if it's incompatible with the first definition's superclass.

It's difficult to do this correctly for ES5-style classes because we don't have a single point that declares the class/properties, but prototype properties potentially scattered throughout the code. This could be easier for ES6-style classes because there is a single CLASS node representing the class declaration. We could check for structural compatibility there.

**Comment by The-Peppester** [comment]
Yes, this does appear to be a problem ([tested here](https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250Avar%2520C%2520%253D%2520class%2520%257B%250A%2520%2520f()%2520%257Bconsole.log(%2522one%2522)%253B%257D%250A%257D%253B%250AC%2520%253D%2520class%2520%257B%250A%2520%2520g()%2520%257Bconsole.log(%2522two%2522)%253B%257D%250A%257D%253B%250Avar%2520n%2520%253D%2520new%2520C()%253B%250A%250An.f()%253B%250An.g()%253B)). I would guess the problem might be something like this: somewhere in the closure compiler source code, classes are additive accumulated. Then, the solution is to instead only use the last class assignment to a variable for its properties instead of accumulating the properties.

**Comment by lauraharker** [comment]
Created Google internal issue https://b.corp.google.com/issues/140735194, since I just ran into this again.

https://closure-compiler-debugger.appspot.com/#input0%3Dclass%2520ParentA%2520%257B%257D%250Aclass%2520ParentB%2520%257B%257D%250A%250Alet%2520Klazz%253B%250A%250Aif%2520(true)%2520%257B%250A%2520%2520Klazz%2520%253D%2520class%2520extends%2520ParentA%2520%257B%257D%253B%250A%257D%2520else%2520%257B%250A%2520%2520Klazz%2520%253D%2520class%2520extends%2520ParentB%2520%257B%257D%253B%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26CHECK_TYPES%3Dtrue%26CLOSURE_PASS%3Dtrue%26PRESERVE_TYPE_ANNOTATIONS%3Dtrue%26PRETTY_PRINT%3Dtrue
